Feature/footer v2

shkim1026 opened the pull request.

shkim1026 committed:
Refactors HTML for Ver2. Removes blurb. Adds two menus. Adds additional schema settings.

shkim1026 committed:
Adds newsletter form. Refactors HTML and CSS to accomodate newsletter inputs (for all media viewports)

shkim1026 committed:
Applys drop down menu to footer menus for mobile viewports

shkim1026 committed:
Rewrites JS drop down script to accordion. Refactors corresponding CSS. Adds chevron icon to footer menu.

shkim1026 committed:
Removes scripts in {%form%} for newsletter

shkim1026 committed:
Adds error and success message on 'newsletter' form submission.

shkim1026 committed:
Fixes newsletter margin/padding for small media viewports

shkim1026 committed:
Adjusts footer menu's column spacing/width. Adjusts newsletter width

davidhonam commented on assets/custom.js:
```diff
@@ -243,4 +243,23 @@ $('.remove').click(function(){
+    const submenu = $(this).next();
+    const scrollHeight = submenu.prop('scrollHeight');
+    const expandedHeight = scrollHeight + "px";
+    const mediaQuery = window.matchMedia('(max-width: 767px'); //Allows footer menu heading to be clicked ONLY on mobile viewports
+    if (mediaQuery.matches) {
```
this is cool. i've never used this. i've always used the jquery way: `$(window).width()` 

ie. (`if ($(window).width() > 767 {}`)

matchMedia seems like a better solution, so i'd stick with it

davidhonam commented on assets/custom.js:
```diff
@@ -243,4 +243,23 @@ $('.remove').click(function(){
+    const mediaQuery = window.matchMedia('(max-width: 767px'); //Allows footer menu heading to be clicked ONLY on mobile viewports
+    if (mediaQuery.matches) {
+        $(this).toggleClass('active');
+        if ($(this).hasClass('active')) {
+            submenu.css('maxHeight', expandedHeight);
+            chevron.css('transform', 'rotateX(180deg)');
+        } else {
+            submenu.css('maxHeight', '0')
+            chevron.css('transform', 'rotateX(0deg)');
```
i think this can be achieved through css: `.footer__links-heading.active` 
(remember, always do styling in CSS if possible)

also, not sure why you would need to set `max-height`. you should be able to achieve the accordion without setting it but let me know if otherwise.

davidhonam replied:
ah i see, you're doing it for the animation. gotcha, nevermind then

davidhonam reviewed (approved):
looks good. i have no notes that i can think of

davidhonam commented:
i do have one note, when hovering over the footer menu items, the cursor should be a pointer

davidhonam commented:
also, let's reduce the newsletter placeholder font size so that it's not cut off "Enter your email address"

shkim1026 committed:
Fixes newsletter placeholder text cut off: Adds min-width to newsletter input. Adds 'cursor:pointer' to newsletter submit input.

shkim1026 commented:
> i do have one note, when hovering over the footer menu items, the cursor should be a pointer

Does it not have a pointer cursor for you? Because it does for me. Added one for the newsletter submit button though.

> also, let's reduce the newsletter placeholder font size so that it's not cut off "Enter your email address"

Gotcha. Added a 'min-width' to the input so the placeholder isn't cut off.
